mobile/library/java/org/chromium/net/impl/CronvoyUrlRequest.java

```diff
@@ -880,15 +866,19 @@ public void onData(ByteBuffer data, boolean endStream, EnvoyStreamIntel streamIn
         return;
       }
 
+      ByteBuffer userBuffer = mUserCurrentReadBuffer;
+      mUserCurrentReadBuffer = null; // Avoid the reference to a potentially large buffer.
+      int dataRead = data.remaining();
+      // It is important to copy the `data` into the `userBuffer` outside the thread execution
+      // because the `data` is backed by a direct `ByteBuffer` and it will be destroyed upon once
+      // the `onData` completes.
```

Comment:
so is this the only place we'll have buffer copies once everything is in the final desired end state?

Reply:
Yup!